### DISCUSSION TITLE
Misunderstanding very simple RLS and Join for "owner id"

### DISCUSSION THREAD

**Discussion by adamhp** [supabase/supabase#7855] [Questions] [answered]
I have several objects in my database that are "owned", where they have foreign keys to another table representing my user.

More specifically, I have the `auth.users` default table with supabase auth, a `profiles` table that is related to that via foreign key, and then all other tables are related to `profiles`.

So, `profiles` maps to `auth.users` with `id` -> `id` FK.
All other tables map to `profiles` with `owner_id` -> `id` FK.

Example:
`profiles`
| id | email          |
|----|----------------|
| 1  | usera@mail.com |
| 2  | userb@mail.com |

`objects`
| owner_id | name     |
|----------|----------|
| 1        | object-a |
| 1        | object-b |
| 2        | object-c |
| 2        | object-d |

I have an RLS policy that applies to ALL operations and checks that 
`(uid() = owner_id)`
 
For whatever reason, this only works on a direct call to the `objects` table:
```javascript
client
    .from('social_accounts')
    .select('*')
    .eq('owner_id', ownerId)
    .then((res) => res.data);
```

But not with join:
```javascript
client
    .from('profiles')
    .select(`
      *,
      objects (*)
    `,)
    .eq('id', ownerId)
    .single()
    .then((res) => res.data);
```

I.e. `objects` returns `[]` in the above. 

What am I missing? An additional filter on the select maybe?

**Comment by nextdev1111** [comment]
I think you should also have a match field for joint. 

In simple words, just pass a filter which says that
`match({
'Ownerid':owner id,
'Objects.ownerid': owner id 
})`

**Comment by adamhp** [comment, marked as answer]
I'm realizing this actually has to do with `supabase-auth-helpers` not actually having a session but returning the `user`. It was confusing because I had a public readable RLS on `profiles`, so that was working, but the others weren't.

My current solution is to wrap this particular page with something like this:
```javascript
export default function DashboardWrapper() {
  const { isLoading, user, error } = useUser();
  return isLoading ? <Spinner /> : <Dashboard user={user} error={error} />;
}
```

So the Dashboard component actually executes the query, but it won't do so until user is actually loaded properly.
